core/src/html/text_format.rs

```diff
@@ -717,7 +717,34 @@ impl FormatSpans {
                             }
 
                             if let Some(size) = attribute(b"size") {
-                                format.size = size.parse().ok();
+                                // if the numer starts with + or -, the size is relative
+                                let (prefix, size) = if size.starts_with(&[b'+', b'-'][..]) {
+                                    (size.at(0) as u8, &size[1..])
+                                } else {
+                                    (0u8, &size[..])
+                                };
+
+                                // text is ignored from the first non-numeric character
+                                // (including the decimal separator)
+                                let first_foreign_char = size
+                                    .find(|c| c < '0' as u16 || c > '9' as u16)
+                                    .unwrap_or(size.len());
+                                let size = &size[0..first_foreign_char];
+
+                                let size: Option<f64> = size.parse().ok();
+
+                                if let Some(size) = size
+                                    .and_then(|size| match prefix {
+                                        b'+' => format.size.map(|last_size| last_size + size),
+                                        b'-' => format.size.map(|last_size| last_size - size),
+                                        _ => Some(size),
+                                    })
+                                    .map(|size| size.clamp(1.0, 127.0))
```

Comment:
Also, as a side note, the font size seems to be an integer, and programmatically may be set to a value outside of the `[1,127]` range. I'll probably tackle this behavior in a near future